config/crd/experimental/gateway.networking.k8s.io_httproutes.yaml

```
@@ -2991,12 +3310,97 @@ spec:
                                       required:
                                       - type
                                       type: object
+                                      x-kubernetes-validations:
+                                      - message: replaceFullPath must be set when
+                                          type is set to 'ReplaceFullPath'
+                                        rule: 'self.type == ''ReplaceFullPath'' ?
+                                          has(self.replaceFullPath) : true'
+                                      - message: type must be 'ReplaceFullPath' when
+                                          replaceFullPath is set
+                                        rule: 'has(self.replaceFullPath) ? self.type
+                                          == ''ReplaceFullPath'' : true'
+                                      - message: replacePrefixMatch must be set when
+                                          type is set to 'ReplacePrefixMatch'
+                                        rule: 'self.type == ''ReplacePrefixMatch''
+                                          ? has(self.replacePrefixMatch) : true'
+                                      - message: type must be 'ReplacePrefixMatch'
+                                          when replacePrefixMatch is set
+                                        rule: 'has(self.replacePrefixMatch) ? self.type
+                                          == ''ReplacePrefixMatch'' : true'
                                   type: object
                               required:
                               - type
                               type: object
+                              x-kubernetes-validations:
+                              - message: filter.RequestHeaderModifier must be nil
+                                  if the HTTPRouteFilter.Type is not RequestHeaderModifier
```

I don't think we've been very consistent with this, but we should try to match the upstream pattern here where field names/paths match what a user would see when using YAML. For example, look at the core validation here: https://github.com/kubernetes/kubernetes/blob/2c6c4566eff972d6c1320b5f8ad795f88c822d09/pkg/apis/core/validation/validation.go#L1041-L1045.

This comment applies throughout the PR, but will not repeat it everywhere to reduce noise.

```suggestion
                              - message: filter.requestHeaderModifier must be nil
                                  if the HTTPRouteFilter.Type is not RequestHeaderModifier
```

---

Sure can do! I've changed all such occurrences. The places which are still kept capitalized have the intent of referring to the enum (like `RequestRedirect`) instead of the field (`requestRedirect`). If in some place, the intent is not clear enough, I can try and reframe the statement.